extension/js/content_scripts/webmail/gmail/gmail-element-replacer.ts

```diff
@@ -589,8 +589,10 @@ export class GmailElementReplacer extends WebmailElementReplacer {
     return from ? Str.parseEmail(from) : undefined;
   };
 
-  private getLastMsgReplyParams = (convoRootEl: JQuery): FactoryReplyParams => {
-    return { replyMsgId: this.determineMsgId($(convoRootEl).find(this.sel.msgInner).last()) };
```

Comment:
This change goes to correct the logic of the existing code. Since the last container of the message to obtain the msgId does not mean to be where the container where the secure compose get's clicked.

What I mean by is, assume there are several existing converstations in an email thread (doesn't matter if its a group message). Assume the user clicks secure reply for a specific email that is placed in the middle order of the converstation, this will access the appropriate container rather than the last container of the conversation of the opened email thread.

Reply:
I understand clearly now. Thanks for reviewing this PR.